ChangeLog.d/remove-deprecated-items.txt

```diff
@@ -0,0 +1,11 @@
+Removals
+   * Remove MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT, now only the 
+     standard version is supported.
```

Comment:
Maybe it's just me that I'm not an expert of DTLS, but here it's not completely clear to me what the "standard version" refers to. I had to check the change in `mbedtls_config.h` to get an idea. Perhaps changing it to:

```suggestion
   * Remove MBEDTLS_SSL_DTLS_CONNECTION_ID_COMPAT. Now only the 
     standard version (defined in RFC 9146) of DTLS connection ID is supported.
```